gpt_index/indices/vector_store/types.py

```diff
@@ -0,0 +1,44 @@
+"""Vector store index types."""
+
+
+from dataclasses import dataclass
+from typing import Any, List, Optional, Protocol
+
+from gpt_index.data_structs.data_structs import Node
+
+
+@dataclass
+class NodeEmbeddingResult:
+    id: str
+    node: Node
+    embedding: List[float]
+    doc_id: str
+
+
+@dataclass
+class VectorStoreQueryResult:
+    nodes: Optional[List[Node]] = None
+    similarities: Optional[List[float]] = None
+    ids: Optional[List[int]] = None
+
+
+class VectorStore(Protocol):
```

Comment:
file structure-wise, i was thinking the `indices` folder would be specifically around index classes, and stuff that doesn't depend on index classes e.g. VectorStore could live in a top-level folder `gpt_index/vector_store`

that way you also don't need the VectorStoreIndex to have a structure like `indices/vector_store/index/base.py`

Reply:
agreed